guides/developer/extending/integrate-your-CI/index.md

```diff
@@ -0,0 +1,126 @@
+In this example, we will query our CI provider to get metadata for build number `2012` and commit id `2626f5f`.
+
+### Getting artifact metadata
+
+Detailed metadata (like commit message, author, timestamp) is visible in the Environments view in the UI, by clicking on an artifact version.
+
+In order to get the metadata information, you'll have to plug in your CI provider.
+`keel` [buildService](https://github.com/spinnaker/keel/blob/master/keel-igor/src/main/kotlin/com/netflix/spinnaker/keel/igor/BuildService.kt) is calling `igor` to fetch this information. It uses the endpoint `<igorBaseURL>/ci/builds`, which looks like:
+```
+  @GET("/ci/builds")
+  @Headers("Accept: application/json")
+  suspend fun getArtifactMetadata(
+    @Query("commitId") commitId: String,
+    @Query("buildNumber") buildNumber: String,
+    @Query("projectKey") projectKey: String? = null,
+    @Query("repoSlug") repoSlug: String? = null,
+    @Query("completionStatus") completionStatus: String? = null
+  ): List<Build>?
+```
+
+At Netflix, we pass as input the `commitId` and `buildNumber`. Those are mandatory.
+The response contains a list of [builds](https://github.com/spinnaker/keel/blob/master/keel-igor/src/main/kotlin/com/netflix/spinnaker/keel/igor/model/Build.kt) (we pick the first from the list), which looks like:
+```
+data class Build(
+  val building: Boolean = false,
+  val fullDisplayName: String? = null,
+  val name: String? = null,
+  val number: Int = 0,
+  val duration: Long? = null,
+  /** String representation of time in nanoseconds since Unix epoch  */
+  val timestamp: String? = null,
+
+  val result: Result? = null,
+  val url: String? = null, //url to get build details from the CI provider (like jenkins)
+  val id: String? = null, //build uid
+
+  val scm: List<GenericGitRevision>? = null,
+  val properties: Map<String, Any?>? = null
+)
+```
+The `build` object is a mirror object of [GenericBuild](https://github.com/spinnaker/igor/blob/master/igor-core/src/main/java/com/netflix/spinnaker/igor/build/model/GenericBuild.java) in `igor`.
+
+
+Note: if you already implemented the `/ci/builds` endpoint as descries above, jump to #3 and you are done!
+
+If you are interested in plugging your CI provider, you'll need to do the following:
+1. Create a fork to `igor`
+2. Implement the `ci/builds` endpoint, and return a `GenericBuild` response (which is converted to `Build` object above).
+- [Controller endpoints](https://github.com/spinnaker/igor/blob/master/igor-web/src/main/java/com/netflix/spinnaker/igor/ci/CiController.java)
+- [Generic build service](https://github.com/spinnaker/igor/blob/master/igor-web/src/main/java/com/netflix/spinnaker/igor/ci/CiBuildService.java)
+- Example: your service implementation should be somthing like:
+```
+ public class YourCompanyNameCIService implements CiBuildService[...]
+```
+3. Make sure the artifacts being sent to `keel` contains the `buildNumber` and `commitId`.
```

Comment:
This is redundant with the section above, right? Consider just adding a link here in a sentence like "If you're interested in plugging in a new CI provider, follow the steps in section [link] above".